docs/products/postgresql/concepts/pg-connection-pooling.rst

```diff
@@ -0,0 +1,92 @@
+PostgreSQL connection pooling
+=============================
+
+Connection pooling in Aiven for PostgreSQL services allows you to maintain very large numbers of connections to a database while minimizing the consumption of server resources.
+
+
+Aiven for PostgreSQL connection pooling uses `PGBouncer <https://www.pgbouncer.org/>`_ to manage the database connection. Each pool can handle up to 5000 database client connections. Unlike when you connect directly to the PostgreSQL server, each client connection does not require a separate backend process on the server. PGBouncer automatically inserts the client queries and only uses a limited number of actual backend connections, leading to lower resource usage on the server and better total performance.
+
+Why connection pooling?
+------------------------
+
+A high number of backend connections can become a problem with PostgreSQL, as the resource cost per connection is quite high due to how PostgreSQL manages client connections. PostgreSQL creates a separate backend process for each connection, and the unnecessary memory usage caused by the processes will start affecting the total throughput of the system at some point. Moreover, if each connection is very active, the performance can be affected by the high number of parallel executing tasks.
+
+It makes sense to have enough connections so that each CPU core on the server has something to do (each connection can only utilise a single CPU core), but a hundred connections per CPU core may be too much. All this is workload-specific, but often a good number of connections to have is roughly 3-5 times the CPU core count.
+
+
+
+.. Note::
+    PostgreSQL 9.6 introduced limited parallelization support for `running queries in parallel <https://www.postgresql.org/docs/9.6/static/parallel-query.html>`_ on multiple CPU cores.
+
+
+Without a connection pooler, the database connections are handled directly by PostgreSQL backend processes, with one process per connection:
+
+.. mermaid::
+
+    graph LR
+        classDef style_main_node fill:orange,stroke:#333,stroke-width:0px;spacing:0px;
```

Comment:
Please don't add styles to diagrams - we can add diagram styles later, and might generate image sets for both dark and light styles at some point.

Reply:
removed styles